pkg/receive/handler.go

```diff
@@ -193,3 +212,80 @@ func ValidateLabels(logger log.Logger, next http.Handler, labels ...string) http
 		next.ServeHTTP(w, r)
 	}
 }
+
+func (h *Handler) TransformWriteRequest(logger log.Logger, next http.Handler) http.HandlerFunc {
+	logger = log.With(logger, "component", "receive", "middleware", "transformWriteRequest")
+	return func(w http.ResponseWriter, r *http.Request) {
+		body, err := ioutil.ReadAll(r.Body)
```

Comment:
This seems to be deprecated  https://pkg.go.dev/io/ioutil#ReadAll

> Deprecated: As of Go 1.16, this function simply calls io.ReadAll.

Reply:
Yeah, but the project was on Go 1.13, so was confused about this. Have changed.